### PR TITLE
Fix DTrace tests

### PR DESCRIPTION
**Pull request by nicomazz**
For a few tests ksh is needed, but we don't have it in FreeBSD by default. As a consequence, I use sh. Many of the DTrace ksh tests are broken in this way, but not all of them.

Then, the preprocessor is missing, so the '-C' option is removed.

A few tests are excluded.

"json" and "usdt" tests are not compiled, because there are dependencies missing.

### CODE REVIEW

File: cddl/usr.sbin/dtrace/tests/common/Makefile [hunk 1]
```diff
@@ -4,6 +4,12 @@
 # FreeBSD, and docsExamples doesn't seem amenable to automated testing.
 
 TESTSDIR=	${TESTSBASE}/cddl/usr.sbin/dtrace/common
+
+
+# A few tests how did not compile in CHERI have been removed from here.
+# The problems were missing dependencies.
+# Tests removed: json usdt
+
```
Comment (qwattash) [inline cddl/usr.sbin/dtrace/tests/common/Makefile:12]: Maybe move them to ``TESTS_SUBDIRS_NOCHERI` to keep building them for other targets?

File: cddl/usr.sbin/dtrace/tests/tools/exclude.sh [hunk 2]
```diff
@@ -210,3 +210,5 @@ exclude EXFAIL common/usdt/tst.user.ksh
 
 # Triggers a lock assertion by using the raise() action from a profile probe.
 exclude SKIP common/ustack/tst.spin.ksh
+
+. $(dirname "$0")/exclude_cheri.sh
```
Comment (nicomazz) [inline cddl/usr.sbin/dtrace/tests/tools/exclude.sh:214]: Which is the best way to do this only if we are in CheriBSD?

File: cddl/usr.sbin/dtrace/tests/tools/exclude_cheri.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+exclude EXFAIL common/aggs/err.D_KEY_TYPE.badkey4.d
+
+# goes into timeout. Probably missing dependencies. Not important.
+exclude SKIP common/aggs/tst.aggpackbanner.ksh
+
+# times are to short. With a faster cpu will probably work.
+exclude EXFAIL common/aggs/tst.clear.d
+exclude EXFAIL common/aggs/tst.cleardenormalize.d
+
+# smal deltas in the output.
```
Comment (nicomazz) [inline cddl/usr.sbin/dtrace/tests/tools/exclude_cheri.sh:19]: I'll fix this

File: cddl/usr.sbin/dtrace/tests/common/Makefile [hunk 4]
```diff
@@ -80,11 +79,17 @@ TESTS_SUBDIRS+= aggs \
...
 		uctf \
 		union \
-		usdt \
 		ustack \
 		vars \
 		version \
 
+# A few tests how did not compile in CHERI have been removed from here.
+# The problems were missing dependencies.
+.if !${TARGET_ARCH:M*cheri*}
```
Comment (arichardson) [inline cddl/usr.sbin/dtrace/tests/common/Makefile:88]: I think this should be `${MACHINE_ABI:Mpurecap}`?

**Comment by gvnn3** [comment]
@nicomazz @arichardson please merge or delete, thanks.

**Comment by jrtc27** [comment]
None of these bits seem relevant, just working around limitations at the time

**Pull request closed without merging**